### PR TITLE
Task-7

### PR DESCRIPTION
**Pull request by lexx129**
**Main scope**
- added `authorization-service` with `token-authorizer` lambda
- updated  `importProductsFile` lambda to use `tokenAuthorizer` lambda as an authorizer
- added getting authorization token from local storage (app should print it to console, if it was found in your LS)

**Additional scope**
- updated axios response handling to alert about 401 and 403 errors

[Direct URL to 'importProductsFile'](https://hrdhhb2s9a.execute-api.eu-west-1.amazonaws.com/dev/import) (for Postman)
[deployed FE URL](https://d1amechee18gmk.cloudfront.net)

### CODE REVIEW

**Comment by AndreyS55** [comment]
Please provide the correct `key` and `value` for localStorage or Front-end repo PR.

File: authorization-service/handlers/tokenAuthorizer.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+export const handler = async (event: APIGatewayTokenAuthorizerEvent, _, callback) => {
+  console.log('Got an event: ', event);
+
+  if(event.type != 'TOKEN'){
+    callback('Unauthorized');
+  }
+
+  try {
+    const token = event.authorizationToken;
```
Comment (AndreyS55) [inline authorization-service/handlers/tokenAuthorizer.ts:18]: I suggest to throw 401 error if `token` not provided

**Comment by lexx129** [comment]
> Please provide the correct `key` and `value` for localStorage or Front-end repo PR.

Why do you need FE source code when there's a link to a deployed version of it?

File: authorization-service/handlers/tokenAuthorizer.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    const encodedCreds = token.split(' ')[1];
```
Comment (AndreyS55) [inline authorization-service/handlers/tokenAuthorizer.ts:19]: I also suggest to check that `token.split(' ')[0] === 'Basic'`

File: authorization-service/handlers/tokenAuthorizer.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            Effect: effect,
+            Resource: event.methodArn,
+          }
+        ],
+      },
+    };
+
+    callback(null, policy);
+  } catch (err){
+    callback(`Unauthorized: ${err.message}`);
```
Comment (AndreyS55) [inline authorization-service/handlers/tokenAuthorizer.ts:45]: Here you should handle 500 errors but not 401

**Comment by AndreyS55** [comment]
> > Please provide the correct `key` and `value` for localStorage or Front-end repo PR.
> 
> Why do you need FE source code when there's a link to a deployed version of it?

Because I don't know what key you used to put token in localStorage. I can check it only in FE repo or you can provide it in PR description. Also I don't know how token should looks on your side, with `Basic` prefix or not.

**Pull request merged**